packages/rpc-core/src/rpc-methods/__tests__/simulate-transaction-test.ts

```diff
@@ -0,0 +1,717 @@
+                        ...message,
+                    ])
+                ).toString('base64') as Base64EncodedWireTransaction,
+                {
+                    accounts: {
+                        addresses: ['DRtXHDgC312wpNdNCSb8vCoXDcofCJcPHdAw4VkJ8L9i' as Base58EncodedAddress],
+                        encoding: 'base64',
+                    },
+                    commitment: 'processed',
+                    encoding: 'base64',
+                }
+            )
+            .send();
+
+        await expect(resultPromise).resolves.toMatchObject({
+            value: expect.objectContaining({
+                accounts: [
+                    expect.objectContaining({
+                        data: ['', 'base64'],
+                    }),
+                ],
+                err: null,
+                logs: expect.any(Array),
+                returnData: null,
+                unitsConsumed: expect.any(BigInt),
+            }),
+        });
+    });
+
+    it('returns account data for a transaction account with base64+zstd encoding', async () => {
+        expect.assertions(1);
+        const [secretKey, { value: latestBlockhash }] = await Promise.all([
+            getSecretKey(),
+            rpc.getLatestBlockhash().send(),
+        ]);
+        const message = getMockTransactionMessage({
+            blockhash: latestBlockhash.blockhash,
+            feePayerAddressBytes: MOCK_PUBLIC_KEY_BYTES,
+            memoString: `Hello from the web3.js tests! [${performance.now()}]`,
+        });
+        const signature = new Uint8Array(await crypto.subtle.sign('Ed25519', secretKey, message));
+        const resultPromise = rpc
+            .simulateTransaction(
+                Buffer.from(
+                    new Uint8Array([
+                        0x01, // Length of signatures
+                        ...signature,
+                        ...message,
+                    ])
+                ).toString('base64') as Base64EncodedWireTransaction,
+                {
+                    accounts: {
+                        addresses: ['DRtXHDgC312wpNdNCSb8vCoXDcofCJcPHdAw4VkJ8L9i' as Base58EncodedAddress],
+                        encoding: 'base64+zstd',
+                    },
+                    commitment: 'processed',
+                    encoding: 'base64',
+                }
+            )
+            .send();
+
+        await expect(resultPromise).resolves.toMatchObject({
+            value: expect.objectContaining({
+                accounts: [
+                    expect.objectContaining({
+                        data: [expect.any(String), 'base64+zstd'],
+                    }),
+                ],
+                err: null,
+                logs: expect.any(Array),
+                returnData: null,
+                unitsConsumed: expect.any(BigInt),
+            }),
+        });
+    });
+
+    it('returns account data for a transaction account with jsonParsed encoding', async () => {
```

Comment:
Add a test for ‘asked for jsonParsed but fell back to base64’?

Reply:
That's actually the test I wrote here, there's no JSON parser for the account I'm testing. The missing one is one that actually does parse as JSON, I'll see how easily I can add that.

Reply:
Got both now!